packages/google_identity_services_web/lib/src/js_interop/package_web_tweaks.dart

```diff
@@ -2,34 +2,75 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// Provides some useful tweaks to `package:web`.
-library package_web_tweaks;
-
 import 'dart:js_interop';
+
 import 'package:web/web.dart' as web;
 
+// TODO(kevmoo): Make this file unnecessary, https://github.com/dart-lang/web/issues/175
+
 /// This extension gives web.window a nullable getter to the `trustedTypes`
 /// property, which needs to be used to check for feature support.
 extension NullableTrustedTypesGetter on web.Window {
```

Comment:
Why are we using extensions? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension

Reply:
Extension types and extensions seem to be The New Way of doing JS-interop, rather than the `@staticInterop` annotation. See this: https://github.com/flutter/flutter/pull/143274 (I followed @kevmoo's lead here)

Reply:
Okay, if Hixie is on board with this use case an an exception to the Flutter style guide that's fine.

Reply:
Should we ask @Hixie directly?

Reply:
Most of the file is the same extension type pattern that he just approved in the other PR, so I was taking that as confirmation that it should be fine.

Reply:
> (This MAY be rewritable as extension type NullableTrustedTypesGetter._(web.Window) implements web.Window... (I'm going to try!))

OK, I tried:

```dart
extension type NullableTrustedTypesGetter._(web.Window _) implements web.Window {
```

instead of

```dart
extension NullableTrustedTypesGetter on web.Window {
```

And the compiler seems to take it, HOWEVER, when using the `web.window` instance of the `web.Window` class, my "wrapper" is not automagically applied, and I cannot access the nullable setter (because it should be of type `NullableTrustedTypesGetter`). This could be fixed by casting `as NullableTrustedTypesGetter` but... IMO that makes the code less legible.

So the rule of thumb seems to be:

* My custom JS-Interop types need to `extension type ... implements JSObject` (as described in the docs, this is [how `class Window` is defined in `package:web`](https://github.com/dart-lang/web/blob/975e55cbe4ac683c33d5256a91f4133fa97b4c34/lib/src/dom/html.dart#L2707-L2709))
* Existing JS-interop types that come from elsewhere and need some "tweaks" (like `web.Window` in this example) need to be a normal `extension ... on pkg.ForeignType`, so it gets applied automatically without us having to cast or doing anything to that foreign object.

(Of course, the rule of thumb will break down when we discover more complex use-cases, so take it with a grain of salt!)